src/main/resources/sql/create.sql

```diff
@@ -0,0 +1,46 @@
+CREATE TABLE IF NOT EXISTS "shop" (
+  "id"      SERIAL PRIMARY KEY,
+  "name"    VARCHAR(255)        NOT NULL,
+  "address" VARCHAR(255) UNIQUE NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS "customer" (
+  "id"      SERIAL PRIMARY KEY,
+  "name"    VARCHAR(255) NOT NULL,
+  "surname" VARCHAR(255) NOT NULL,
+  "age"     INT2         NOT NULL,
+  "shop_id" INT          NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS "purchase" (
+  "id"          SERIAL PRIMARY KEY,
+  "date"        TIMESTAMP NOT NULL,
+  "customer_id" INT       NOT NULL,
+  UNIQUE ("date","customer_id")
```

Comment:
Why you think that this constraint is needed?

Reply:
As for me, there no way for the same user to provide few purchases at the same time... I was even thought to use this pair as a primary key. Is there smth incorrect?